src/modm/board/nucleo_g431kb/board.hpp

```diff
@@ -0,0 +1,164 @@
+	static constexpr uint32_t Apb1      = Frequency;
+	static constexpr uint32_t Apb2      = Frequency;
+
+	static constexpr uint32_t Cordic    = Ahb1;
+	static constexpr uint32_t Crc       = Ahb1;
+	static constexpr uint32_t Dma       = Ahb1;
+	static constexpr uint32_t Dma1      = Dma;
+	static constexpr uint32_t Dma2      = Dma;
+	static constexpr uint32_t DmaMux    = Dma;
+	static constexpr uint32_t Fmac      = Ahb1;
+
+	static constexpr uint32_t Adc       = Ahb2;
+	static constexpr uint32_t Adc1      = Adc;
+	static constexpr uint32_t Adc2      = Adc;
+	static constexpr uint32_t Dac       = Ahb2;
+	static constexpr uint32_t Dac1      = Dac;
+	static constexpr uint32_t Dac3      = Dac;
+	static constexpr uint32_t Rng       = Ahb2;
+
+	static constexpr uint32_t Can       = Apb1;
+	static constexpr uint32_t Fdcan1    = Can;
+	static constexpr uint32_t I2c       = Apb1;
+	static constexpr uint32_t I2c1      = I2c;
+	static constexpr uint32_t I2c2      = I2c;
+	static constexpr uint32_t I2c3      = I2c;
+	static constexpr uint32_t I2c4      = I2c;
+	static constexpr uint32_t Lptim     = Apb1;
+	static constexpr uint32_t Lpuart    = Apb1;
+	static constexpr uint32_t Rtc       = Apb1;
+	static constexpr uint32_t Spi2      = Apb1;
+	static constexpr uint32_t Spi3      = Apb1;
+	static constexpr uint32_t Uart4     = Apb1;
+	static constexpr uint32_t Uart5     = Apb1;
+	static constexpr uint32_t Usart2    = Apb1;
+	static constexpr uint32_t Usb       = Apb1;
+	static constexpr uint32_t Apb1Timer = Apb1 * 1;
+	static constexpr uint32_t Timer2    = Apb1Timer;
+	static constexpr uint32_t Timer3    = Apb1Timer;
+	static constexpr uint32_t Timer4    = Apb1Timer;
+	static constexpr uint32_t Timer5    = Apb1Timer;
+	static constexpr uint32_t Timer6    = Apb1Timer;
+	static constexpr uint32_t Timer7    = Apb1Timer;
+
+	static constexpr uint32_t Sai1      = Apb2;
+	static constexpr uint32_t Spi1      = Apb2;
+	static constexpr uint32_t Usart1    = Apb2;
+	static constexpr uint32_t Apb2Timer = Apb2 * 1;
+	static constexpr uint32_t Timer1    = Apb2Timer;
+	static constexpr uint32_t Timer8    = Apb2Timer;
+	static constexpr uint32_t Timer15   = Apb2Timer;
+	static constexpr uint32_t Timer16   = Apb2Timer;
+	static constexpr uint32_t Timer17   = Apb2Timer;
+	static constexpr uint32_t Timer20   = Apb2Timer;
+
+	static bool inline
+	enable()
+	{
+		Rcc::enableInternalClock();	// 16MHz
+		Rcc::PllFactors pllFactors{
+			.pllM = 4,	//  16MHz / M= 4 ->   4MHz
+			.pllN = 85,	//   4MHz * N=85 -> 340MHz
+			.pllR = 2,	// 336MHz / R= 2 -> 170MHz = F_cpu
+		};
+		Rcc::enablePll(Rcc::PllSource::InternalClock, pllFactors);
+		Rcc::setFlashLatency<Frequency>();
+		// switch system clock to PLL output
+		Rcc::enableSystemClock(Rcc::SystemClockSource::Pll);
+		Rcc::setAhbPrescaler(Rcc::AhbPrescaler::Div1);
+		// APB1 has max. 170MHz
+		Rcc::setApb1Prescaler(Rcc::Apb1Prescaler::Div1);
+		Rcc::setApb2Prescaler(Rcc::Apb2Prescaler::Div1);
```

Comment:
Is this correct? One of these typically has a lower max frequency, usually ≤f_cpu/2.

Reply:
I believe it is. Datasheet refers to them as low-speed and high-speed APB buses, but says max frequency on both is 170MHz.

Reply:
CubeMX says yes:

![grafik](https://user-images.githubusercontent.com/2820734/99198013-21c3f480-2796-11eb-8d1c-b0f02fe90b87.png)